compiler/rustc_mir_transform/src/match_branches.rs

```diff
@@ -48,6 +54,10 @@ impl<'tcx> crate::MirPass<'tcx> for MatchBranchSimplification {
                 should_cleanup = true;
                 continue;
             }
+            if simplify_to_copy(tcx, body, bb_idx, param_env).is_some() {
+                should_cleanup = true;
+                continue;
+            }
```

Comment:
Should this be made a standalone MirPass? This will be easier to have some analyses computed only once.